crates/swc_ecma_lints/src/rules/no_console.rs

```diff
@@ -60,7 +65,19 @@ impl NoConsole {
 impl Visit for NoConsole {
     noop_visit_type!();
 
-    fn visit_ident(&mut self, id: &Ident) {
-        self.check(id);
+    fn visit_member_expr(&mut self, member: &MemberExpr) {
+        if let Expr::Ident(ident) = member.obj.as_ref() {
+            match &member.prop {
+                MemberProp::Ident(Ident { sym, .. }) => {
+                    self.check(member.span, ident, &*sym);
+                }
+                MemberProp::Computed(ComputedPropName { expr, .. }) => {
+                    if let Expr::Lit(Lit::Str(Str { value, .. })) = expr.as_ref() {
+                        self.check(member.span, ident, &*value);
+                    }
```

Comment:
I think we need `else { expr.visit_with(self); }` as `console.log` can be used as a property of a computed member expression.